flink-table/flink-table-runtime/src/main/java/org/apache/flink/table/runtime/functions/table/lookup/fullcache/inputformat/InputFormatCacheLoader.java

```diff
@@ -107,7 +109,11 @@ protected void reloadCache() throws Exception {
         } catch (InterruptedException ignored) { // we use interrupt to close reload thread
         } finally {
             if (cacheLoadTaskService != null) {
+                // if main cache reload thread encountered an exception,
+                // it interrupts underlying InputSplitCacheLoadTasks threads
                 cacheLoadTaskService.shutdownNow();
```

Comment:
Out of curiosity: Was it a deliberate decision to rely on the commonForkPool (see [ReloadTriggerContext:52](https://github.com/apache/flink/blob/10a6f41fa12284f15af55c359c0c0954800f02de/flink-table/flink-table-runtime/src/main/java/org/apache/flink/table/runtime/functions/table/lookup/fullcache/ReloadTriggerContext.java#L52) if `numSplits == 1` and create a dedicated thread pool if we have more than 1 split? I'm just wondering whether aligning both would be an option by relying on `CompletableFuture.runAsync` here as well and, that way, avoiding to wait for the shutdown.

I couldn't find any reasoning to do that other than readability/less code.

Reply:
To be honest, during development I didn't think about relying on ForkJoinPool. As I understand, it's very useful when we have many little tasks, that also can be created in already running tasks. But in our case we have fixed amount ( = number of splits) of long running tasks, that won't create other subtasks. So simple fixed thread pool looked as a pretty straightforward way to implement cache loading. Plus, if we will use commonForkPool, there can be a situation when we will utilize all threads in this pool (our tasks are long-running), and other tasks that relies on commonForkPool with be starving, which is undesirable behavior, as I understand. Correct me, if I'm wrong, please

Reply:
I'm not sure what the question is.

We certainly shouldn't use the common fork pool. I can see what caused that decision to be made, but I'd say the wrong conclusion was drawn. reloadCache should've explictly been an async operation. That would also make it obvious when looking at the class where concurrency actually exists.

If this operation is so heavy that you don't want to call it in the calling thread then it probably shouldn't run in the common pool.

Not a huge fan of repeatedly creating a new executor service btw; I'd rather create it once for a consistent resource profile. As a side-effect it would then also be trivial to make this call async.

Reply:
yeah, I was just curious about your opinion on the usage of common pool. It felt like implementing it asynchronously would have worked here. I just struggled to think of a better way to enable thread loading based on the number of `InputSplits` to optimize the resource utilization per cache loading cycle.

Reply:
In my latest commit I changed `CacheLoader`, so now it returns future directly + it doesn't rely on common ForkJoinPool. Have a look, please.